onnxruntime/core/optimizer/matmul_transpose_fusion.cc

```diff
@@ -142,13 +142,14 @@ static Node* ReorderCastAndTranspose(Graph& graph, Node* cast,
   const std::vector<NodeArg*> new_transpose_input_defs = {&new_cast_output};
   const std::vector<NodeArg*> new_transpose_output_defs = {cast_output};
 
-  (void)graph.AddNode(graph.GenerateNodeName(cast->Name() + "_transformed"),
+  Node& new_cast = graph.AddNode(graph.GenerateNodeName(cast->Name() + "_transformed"),
```

Comment:
This is for handling Transpose->Cast pattern 

but we don't see such pattern in the target model 


![image](https://user-images.githubusercontent.com/9906745/131559292-35e1e99d-672c-40ab-ac03-f41c84afd8fa.png)

Reply:
Did this fix address the issue above?

Reply:
The erroring pattern is this:
![image](https://user-images.githubusercontent.com/14295305/131560507-2dd9e395-134f-4d14-9a4f-df325db99de4.png)

and this PR solves the error:
`onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for the node Cast_826_transformed:Cast(13)`

Reply:
got it, signed off.